wallets/react-web3wallet/README.md

```diff
@@ -0,0 +1,261 @@
+
+1. Go to [WalletConnect Cloud](https://cloud.walletconnect.com/sign-in) and obtain a project id
+
+2. Add your project details in [WalletConnectUtil.ts](https://github.com/WalletConnect/web-examples/blob/main/wallets/react-wallet-v2/src/utils/WalletConnectUtil.ts) file
+
+3. Install dependencies `yarn install` or `npm install`
+
+4. Setup your environment variables
+
+```bash
+cp .env.local.example .env.local
+```
+
+Your `.env.local` now contains the following environment variables:
+
+- `NEXT_PUBLIC_PROJECT_ID` (placeholder) - You can generate your own ProjectId at https://cloud.walletconnect.com
+- `NEXT_PUBLIC_RELAY_URL` (already set)
+
+5. Run `yarn dev` or `npm run dev` to start local development
+
+## Navigating through example
+
+1. Initial setup and initializations happen in [_app.ts](https://github.com/WalletConnect/web-examples/blob/main/wallets/react-wallet-v2/src/pages/_app.tsx) file
+2. Web3Wallet client, ethers and cosmos wallets are initialized in [useInitialization.ts ](https://github.com/WalletConnect/web-examples/blob/main/wallets/react-wallet-v2/src/hooks/useInitialization.ts) hook
+3. Subscription and handling of WalletConnect events happens in [useWalletConnectEventsManager.ts](https://github.com/WalletConnect/web-examples/blob/main/wallets/react-wallet-v2/src/hooks/useWalletConnectEventsManager.ts) hook, that opens related [Modal views](https://github.com/WalletConnect/web-examples/tree/main/wallets/react-wallet-v2/src/views) and passes them all necessary data
+4. [Modal views](https://github.com/WalletConnect/web-examples/tree/main/wallets/react-wallet-v2/src/views) are responsible for data display and handling approval or rejection actions
+5. Upon approval or rejection, modals pass the request data to [RequestHandlerUtil.ts](https://github.com/WalletConnect/web-examples/blob/main/wallets/react-wallet-v2/src/utils/RequestHandlerUtil.ts) that performs all necessary work based on the request method and returns formated json rpc result data that can be then used for WalletConnect client responses
+
+## Migrate from `sign-client` to `Web3Wallet`
+
+1. Initialization
+```javascript
+    // metadata of your app
+    const metadata = {
+        name: "Demo app",
+        description: "Demo Client as Wallet/Peer",
+        url: "www.walletconnect.com",
+        icons: [],
+    };
+    
+    /* old */
+    import { SignClient } from "@walletconnect/sign-client";
+
+    const signClient = await SignClient.init({
+        projectId: process.env.NEXT_PUBLIC_PROJECT_ID,
+        metadata,
+    })
+
+    /* new */
+    import { Core } from "@walletconnect/core";
+    import { Web3Wallet } from "@walletconnect/web3wallet";
+
+    const core = new Core({
+        projectId: process.env.NEXT_PUBLIC_PROJECT_ID,
+    })
+
+    const web3wallet = await Web3Wallet.init({
+        core, // <- pass the shared `core` instance
```

Comment:
Is passing the core manually always required atm? Could we abstract that step too as part of the Web3Wallet wrapper or am I missing something?

Reply:
https://gist.github.com/pedrouid/4067a262550ffb80ea93ab21cedba2fb#requirements
From the requirements
`Web3Wallet MUST require the Core Client in init`
I'm understanding that it must be provided

Reply:
Gotcha, makes sense to adhere to the spec here if that's how it's laid out 👌 Something to think about in terms of possible simplifications/doing this internally automatically if no core provided similar to how the individual clients do this.

Reply:
Will bring it up during Sign/Auth sync 👍